history/report_utils.py

```diff
@@ -526,13 +540,110 @@ def agg_liability_table(data, tax):
     cur_df.index = ["Current"]
     new_df.index = ["New"]
     final_df = pd.concat([cur_df, new_df])
-    diff = final_df.loc["Current"] - final_df.loc["New"]
+    diff = final_df.loc["New"] - final_df.loc["Current"]
     final_df.loc["Change"] = diff.round(1)
     pct_change = diff / final_df.loc["Current"] * 100
     final_df.loc["Pct Change"] = pct_change.round(2)
     return final_df.to_markdown()
 
 
+def projection_table(data, category):
+    """
+    Creates a markdown table to display detailed projections
+    """
+    df = data[data["Category"].str.contains(category)].copy()
+    cur_df = df[df["Category"] == f"Current {category}"].copy()
+    new_df = df[df["Category"] == f"New {category}"].copy()
+    cur_df.drop("Category", axis=1, inplace=True)
+    new_df.drop("Category", axis=1, inplace=True)
+    cur_df = cur_df.set_index("Year").transpose().round(1)
+    new_df = new_df.set_index("Year").transpose().round(1)
+    diff = new_df - cur_df
+    diff = diff.round(1)
+    pct_change = diff / cur_df * 100
+    pct_change = pct_change.round(2)
+    cur_df.index = ["Current"]
+    new_df.index = ["New"]
+    diff.index = ["Change"]
+    pct_change.index = ["Pct Change"]
+    final_df = pd.concat([cur_df, new_df, diff, pct_change])
+
+    return final_df.to_markdown()
+
+
+def calculate_agi_share(calc, year):
+
+    calc.advance_to_year(year)
+    calc.calc_all()
+
+    vdf = calc.dataframe(['s006', 'c00100', 'expanded_income']).sort_values(by='expanded_income', ascending=False)
```

Comment:
Line length looks quite long here.  You should probably wrap to 90 or less characters per line.